### PR TITLE
[testclient] Printing aggregated data when client exit

### PR DESCRIPTION
**Pull request by yuruguo**
### Motivation
Printing aggregated data includes throughput and delay when the client exits in `PerformanceClient` / `ManagedLedgerWriter`

### Modifications
- Add `printAggregatedStats` method to print delay data in class `PerformanceClient`
- Add a ShutdownHook that call `printAggregatedThroughput` and `printAggregatedStats` in class `PerformanceClient`
- Add `printAggregatedThroughput` into ShutdownHook in class `ManagedLedgerWriter`

### Documentation
- [x] no-need-doc

### CODE REVIEW

**Comment by yuruguo** [comment]
PTAL @merlimat @eolivelli @codelipenghui @hangc0276 @315157973 @gaoran10 THX!

File: pulsar-testclient/src/main/java/org/apache/pulsar/proxy/socket/client/SimpleTestProducerSocket.java [hunk 1]
```diff
@@ -45,6 +45,7 @@
 @WebSocket(maxTextMessageSize = 64 * 1024)
 public class SimpleTestProducerSocket {
     public static Recorder recorder = new Recorder(TimeUnit.SECONDS.toMillis(120000), 5);
+    public static Recorder cumulativeRecorder = new Recorder(TimeUnit.SECONDS.toMillis(120000), 5);
```
Comment (gaoran10) [inline pulsar-testclient/src/main/java/org/apache/pulsar/proxy/socket/client/SimpleTestProducerSocket.java:48]: I'm not sure why need to add a new field `cumulativeRecorder`.

Comment (yuruguo) [reply pulsar-testclient/src/main/java/org/apache/pulsar/proxy/socket/client/SimpleTestProducerSocket.java:48]: Consistent with other implementations such as `PerformanceProducer`, `PerformanceReader`, `ManagedLedgerWriter` and `PerformanceConsumer`

Comment (congbobo184) [reply pulsar-testclient/src/main/java/org/apache/pulsar/proxy/socket/client/SimpleTestProducerSocket.java:48]: It doesn’t seem to be used in `SimpleTestProducerSocket`

Comment (yuruguo) [reply pulsar-testclient/src/main/java/org/apache/pulsar/proxy/socket/client/SimpleTestProducerSocket.java:48]: I will remove this variable `cumulativeRecorder` and directly reuse `recorder` to get the aggregated data. @gaoran10 @congbobo184

**Comment by yuruguo** [comment]
ptal @congbobo184 @zymap @gaoran10

**Comment by yuruguo** [comment]
PTAL @congbobo184 @zymap  @gaoran10 ,thx!

**Comment by yuruguo** [comment]
PTAL @merlimat @eolivelli @congbobo184 @zymap @gaoran10 ,thx!

**Comment by yuruguo** [comment]
PTAL @merlimat @eolivelli @congbobo184 @zymap @hangc0276 @gaoran10 , thx!

**Review by eolivelli — APPROVED** [review]
LGTM

**Comment by yuruguo** [comment]
PTAL @BewareMyPower thx!

**Review by BewareMyPower — APPROVED** [review]

File: pulsar-testclient/src/main/java/org/apache/pulsar/proxy/socket/client/SimpleTestProducerSocket.java (hunk 1 above)
Comment (BewareMyPower) [reply pulsar-testclient/src/main/java/org/apache/pulsar/proxy/socket/client/SimpleTestProducerSocket.java]: @congbobo184 @gaoran10 Could you take a look again?

**Comment by yuruguo** [comment]
@merlimat @lhotari @codelipenghui @congbobo184 @zymap @gaoran10 PTAL

**Pull request merged**